waiter/src/waiter/cookie_support.clj

```diff
@@ -19,35 +19,19 @@
             [taoensso.nippy :as nippy]
             [waiter.utils :as utils])
   (:import clojure.lang.ExceptionInfo
+           java.net.HttpCookie
            java.nio.charset.StandardCharsets
            org.eclipse.jetty.util.UrlEncoded))
 
-(defn url-decode
```

Comment:
So the 50% timing for parsing the cookies with ring vs the old `cookie-value` call was 223 us vs 16 us, or 10x slower. I'm not confident in writing a 100% correct regex, but a 99% correct regex is certainly doable. The question is whether it's worth the performance cost.